File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoReleasesFragmentModule.java
package mb00.android.codehub.ui.repo.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.repo.interactor.RepoInteractor;
import mb00.android.codehub.ui.repo.router.RepoRouter;
import mb00.android.codehub.ui.repo.viewmodel.RepoReleasesViewModel;


@Module
public class RepoReleasesFragmentModule {

    @Provides
    RepoReleasesViewModel provideRepoReleasesViewModel(RepoInteractor interactor, RepoRouter router) {
        return new RepoReleasesViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoReleasesFragmentModule.kt
package mb00.android.codehub.ui.repo.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.repo.interactor.RepoInteractor
import mb00.android.codehub.ui.repo.router.RepoRouter
import mb00.android.codehub.ui.repo.viewmodel.RepoReleasesViewModel


@Module
class RepoReleasesFragmentModule {

    @Provides
    internal fun provideRepoReleasesViewModel(interactor: RepoInteractor, router: RepoRouter): RepoReleasesViewModel {
        return RepoReleasesViewModel(interactor, router)
    }

}
